### PR TITLE
changes for MESHSYNC persistence

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
So cool, @kumarabd. 

Way to wrangle this component of the architecture from soup to nuts. 🥳

File: router/server.go [hunk 1]
```diff
@@ -122,6 +122,8 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
 	// meshsync endpoint
 	gMux.Handle("/api/v1/meshsync", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.MeshSyncHandler))))
```
Comment (kalradev) [inline router/server.go:123]: We can remove this

**Comment by leecalcote** [comment]
@dependabot rebase

**Comment by leecalcote** [comment]
A couple conflicts to address, @kumarabd

**Comment by leecalcote** [comment]
@kalradev are we ready to merge this PR or do we have some work left?

**Pull request merged**